pandas/io/excel/_base.py

```diff
@@ -910,3 +910,13 @@ def __exit__(self, exc_type, exc_value, traceback):
     def __del__(self):
         # Ensure we don't leak file descriptors
         self.close()
+        if self.engine == "openpyxl":
+            # openpyxl 2.5.5 and earlier could leave open files behind
+            #  https://bitbucket.org/openpyxl/openpyxl/issues/832
+            import openpyxl
+            from distutils.version import LooseVersion
+
+            if LooseVersion(openpyxl.__version__) <= LooseVersion("2.5.5"):
+                import gc
```

Comment:
really this works? this is *very* weird to do *inside* a ``__del__``. 

reading the link I would rather so something like (and only in the openpyxl engine)
```
if LooseVersion(openpyxl.__version__) <= LooseVersion("2.5.5"):
     for s in self.sheets.values:
            s.xml_source.close()
```

Reply:
maybe need a try/except around the close

Reply:
it does work, agreed that calling gc is ugly, will try the xml_source.close thing.

Reply:
no dice, no `sheets` attribute

Reply:
right i am not sure of the correct attribute here
but shoulda be clear if step into it

just really want to avoid the gc call in del

Reply:
@WillAyd any thoughts on how to make this work without gc call?

Reply:
Can we move this cleanup to a test fixture used for the relevant tests? And just ensure that it's called for the relevant openpyxl version before `check_for_file_leaks` is run?

Reply:
I guess as a general comment can we move this to the `_openpyxl` instead of checking the engine in the base class? Or does that not work?

I think you can just use the `get_sheet_by_index` property to iterate over the sheets and close the related xml_source

Reply:
Hmm ignore comment before on moving to `_openpyxl` - I see this is for the `ExcelFile` class and not the `Reader`

I think you could do something like:

```python
sheet_count = len(self._reader.sheet_names)
for i in range(sheet_count):
    sheet = self._reader.get_sheet_by_index(i)
    sheet.xml_source.close()
```

IIUC

Reply:
> sheet.xml_source.close()

still no dice.  I'm sure there is some amount of digging around that can effectively re-implement the fix that openpyxl did, but im not wild about digging into openpyxl internals, especially for a known-fixed bug